### ISSUE TITLE
 DEBUG    : map::unboard_vehicle: vehicle not found

### ISSUE THREAD

**Issue by Androwski3000** [CleverRaven/Cataclysm-DDA#25309]
**Describe the bug**  
I loaded into the savegame and my vehicle is gone. The bug occured after updating from 7811 to 7817.

 DEBUG    : map::unboard_vehicle: vehicle not found
 
 FUNCTION : void map::unboard_vehicle(const tripoint&)
 FILE     : src/map.cpp
 LINE     : 859

Ver: 0.C-31410-g1ac088a (tiles)
Build 7817

EDIT: On restoring to 7811 the same bug occures. [This is like really bad. I made a gipsy run in that game and all my things where in that custom RV.]

**Comment by Androwski3000** [comment]
![vehicle bug 1](https://user-images.githubusercontent.com/42423698/44955031-df5fb700-aeac-11e8-9520-4711a49e9d26.jpg)
The guy stands on the road where the bed of the vehicle was.

**Comment by Androwski3000** [comment]
If there is anything more which I could provide to get this bug fixed, I´m open for suggestions. I had so much fun in that game but it all depended on that vehicle.

**Comment by Androwski3000** [comment]
I stuck my nose into how vehicle IDs are saved for "remembered" vehicles. This information is saved in \save\your map name here\o.0.0
In the format
`,"tracked_vehicles":[{"id":1,"name":"Mobile Meth Lab","x":68,"y":48},`

I compared the logged vehicles with what I can remember from the game. Ingame I noticed pretty late that my custom vehicle is not tracked yet. So I tracked it. It was a modoified Humvee. But I never changed the name on it via "e" in the vehicle menu. So the name was still Humvee.
On going on with playing, the last vehicle I marked for tracking was a Humvee in the S of the map. It should have the ID 57. **But somehow this Humvee kicked my "custom RV Humvee" out of it´s position in the tracking list.** The last marked vehicle has now ID 52 but should have ID 57.

I could find that out by looking at the map coordinates in the tracking list and comparing them with the map and from what I remember. There is now way that I marked the most south vehicle before the others. Just look at the y coordinate. With 110 it is more south then the ID´s 53, 54, 55, 56.

,"tracked_vehicles":[{"id":1,"name":"Mobile Meth Lab","x":68,"y":48},{"id":2,"name":"Security Van","x":67,"y":49},{"id":3,"name":"Car","x":73,"y":45},{"id":4,"name":"Cube Van","x":66,"y":41},{"id":5,"name":"Car","x":66,"y":42},{"id":6,"name":"Fire Truck","x":69,"y":45},{"id":7,"name":"Shopping Cart","x":74,"y":44},{"id":8,"name":"Car","x":75,"y":43},{"id":10,"name":"Pickup Truck","x":71,"y":40},{"id":11,"name":"Shopping Cart","x":66,"y":49},{"id":12,"name":"Wreckage","x":72,"y":41},{"id":14,"name":"Ambulance","x":79,"y":45},{"id":16,"name":"Atomic Mini-Tank","x":75,"y":38},{"id":17,"name":"Heavy Duty Cargo Truck","x":75,"y":37},{"id":18,"name":"Humvee","x":75,"y":36},{"id":19,"name":"Electric Car","x":74,"y":35},{"id":20,"name":"SWAT Truck","x":75,"y":20},{"id":21,"name":"Shopping Cart","x":67,"y":49},{"id":22,"name":"AH-64 Apache Wreckage","x":42,"y":21},{"id":23,"name":"Self-Propelled Howitzer","x":128,"y":16},{"id":24,"name":"Light Tank","x":104,"y":15},{"id":25,"name":"Armored Personnel Carrier","x":104,"y":14},{"id":26,"name":"AH-64 Apache Wreckage","x":85,"y":16},{"id":27,"name":"Car","x":66,"y":48},{"id":28,"name":"Electric SUV","x":28,"y":60},{"id":30,"name":"Hatchback","x":30,"y":58},{"id":31,"name":"Hippie Van","x":74,"y":45},{"id":32,"name":"Flatbed Truck","x":8,"y":30},{"id":33,"name":"Shopping Cart","x":3,"y":29},{"id":34,"name":"Trencher","x":13,"y":33},{"id":35,"name":"Security Van","x":17,"y":31},{"id":36,"name":"Wreckage","x":18,"y":29},{"id":37,"name":"Electric Car","x":18,"y":29},{"id":38,"name":"Armored Personnel Carrier","x":16,"y":29},{"id":39,"name":"Security Van","x":14,"y":29},{"id":40,"name":"Flatbed Truck","x":13,"y":29},{"id":41,"name":"RV","x":14,"y":34},{"id":42,"name":"Wreckage","x":34,"y":58},{"id":43,"name":"Mobile Gun System","x":40,"y":54},{"id":44,"name":"RV","x":30,"y":52},{"id":45,"name":"Wreckage","x":32,"y":59},{"id":46,"name":"Bandit Bulldozer","x":32,"y":66},{"id":47,"name":"UH-60 Wreckage","x":101,"y":37},{"id":48,"name":"Hippie Van","x":66,"y":48},{"id":49,"name":"Security Van","x":66,"y":48},{"id":50,"name":"AH-64 Apache Wreckage","x":1,"y":26},{"id":51,"name":"Armored Personnel Carrier","x":11,"y":27},{"**id":52,"name":"Humvee","x":55,"y":110**},{"id":53,"name":"RV","x":57,"y":75},{"id":54,"name":"AH-64 Apache Wreckage","x":55,"y":82},{"id":55,"name":"Military Cargo Truck","x":51,"y":107},{"id":56,"name":"Armored Car","x":51,"y":105}]

![1](https://user-images.githubusercontent.com/42423698/44997108-3c8e6200-afac-11e8-93d6-c58efb054f8c.jpg)

**Comment by Androwski3000** [comment]
I looked at vehicle coordinates of tracked vehicles in \save\your map name here\o.0.0. That way I found the missing vehicle at another location.

I died because of pulping a stack of zombies on one tile. Which took longer than ecspected and there where still enemies around. I CRTL+ALT+DEL to kill the game before the death was saved.
Somehow vehicle positions are not bound to a savegame but are rather be handled mapwise. So the vehicle was not at the position of the last save. It was instead at it´s position when the game was killed.

Timeline:
...............1.......................2.....................3....................4.................5................>
1. Sleep + Save,
2. Going on with  exploring/travelling,
3. Leaving vehicle for fighting,
4. dying because of pulping a stack of zombies on one tile,
5. CRTL+ALT+DEL before saving the death

![2](https://user-images.githubusercontent.com/42423698/44997567-a52b0e00-afaf-11e8-8109-5e8845626275.jpg)
Picture: The vehicle should habe been where "@" is. That is the position of the save. Instead it was far more S at it´s parking position, when the game was killed via CRTL+ALT+DEL.
